pkg/signup/signup_callback_handler.go

```diff
@@ -0,0 +1,34 @@
+package signup
+
+import (
+	"encoding/json"
+	"github.com/codeready-toolchain/registration-service/pkg/configuration"
+	"github.com/gin-gonic/gin"
+	"log"
+	"net/http"
+)
+
+// SignupCallbackService implements the signup callback endpoint.
+type SignupCallbackService struct {
+	config *configuration.Registry
+	logger *log.Logger
+}
+
+// NewSignupCallbackService returns a new SignupCallbackService instance.
+func NewSignupCallbackService(logger *log.Logger, config *configuration.Registry) *SignupCallbackService {
```

Comment:
What this callback endpoint is about? Is it for checking the signup status? Why you call it "callback"? IMO it's a bit confusing name.

What if we use `POST /signup` for creating a signup/registering a new user and `GET /signup` for checking the signup status?

Reply:
i agree with Alexey. Maybe we could call this `NewSignupService`. And below have the endpoints as `/signup` with the appropriate http verbs.

Reply:
agreed with @alexeykazakov we can remove the callback files and just add post/get and handle both scenarios withint he signup_handler.go

Reply:
Michael and I discussed and agreed on these during our meeting on Friday.  The `/signup` endpoint is invoked as part of the landing page, it produces a redirect URL that sends the user to the RHD SSO to authenticate.  The `/signup-callback` endpoint is where the RHD login redirects the user to, it is this endpoint that will do the work of creating the `UserSignup` resource in the host cluster.  We will then have another endpoint, `check-status` which can be used to check the status of the user's provisioning request.

Reply:
i think we should consider renaming the endpoints. isn't it possible that the user is just signing in and not necessarily signing up? also restful endpoints are meant to point to resources. so perhaps a better endpoint for /signup would be /users and a better endpoint for /signup-callback would be /usersignup since that is the resource we are creating..

Reply:
Why we would need an endpoint for redirecting user to RHD in our backend at all?

Those `signup` and `check-status` endpoints were designed to be use to manage signups on the cluster side. This is why I suggested to actually renamed them to `POST /signup` and  `GET /signup`. It's all about SignUp resource management.

The redirect-callback concept is new to me and I'm not quite understand what it would give us.

Could you please elaborate on that?

cc: @michaelkleinhenz

Reply:
after talking with @alexeykazakov we will need to remove the /signup-callback url. i think we should also rename the endpoint to /usersignup to avoid any confusion and make it clear that we are only handling the usersignup resource rather than the entire signup flow in the backend...

background: 
1. we want the registration app to be stateless, so we don't want to have to add back-end storage/database
2. because of [1.] we are planning to do the redirect in the ui so the flow is as follows: 
            2a. ui makes the call to rhd sso with a redirect url back to the ui
            2b. rhd sso will signup/login the user and provide back a code
            2c. ui makes another call to rhd sso via ajax with the code
            2d. rhd returns back to ui with the token
            2e. ui makes a call to /usersignup with token 
            2f. create usersignup resource and return with appropriate status code

2a-2d is handled by the javascript library.